Error while building SlicerSALT in docker

hi, 

I am trying to build SlicerSALT using the qt5-centos7 Slicer build environment Docker file but the compilation ends with errors. The errors occur while building SRep.

@jcfr, @phcerdan, do you know what could be the problem?

Here is the error I got:

```
[ 74%] Building CXX object E/SRep/SkeletalRepresentationInitializer/Logic/CMakeFiles/vtkSlicerSkeletalRepresentationInitializerModuleLogicPythonD.dir/vtkBackwardFlowLogicPython.cxx.o

/osm_qSlicerUnitsModuleWidgets_module_init.cpp.o
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx: In function ‘PyObject* PyvtkBackwardFlowLogic_runApplyTPS(PyObject*, PyObject*)’:
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx:25:68: error: invalid static_cast from type ‘vtkObjectBase*’ to type ‘vtkBackwardFlowLogic*’
   vtkBackwardFlowLogic *op = static_cast<vtkBackwardFlowLogic *>(vp);
                                                                    ^
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx: In function ‘PyObject* PyvtkBackwardFlowLogic_computePairwiseTPS(PyObject*, PyObject*)’:
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx:55:68: error: invalid static_cast from type ‘vtkObjectBase*’ to type ‘vtkBackwardFlowLogic*’
   vtkBackwardFlowLogic *op = static_cast<vtkBackwardFlowLogic *>(vp);
                                                                    ^
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx: In function ‘PyObject* PyvtkBackwardFlowLogic_generateEllipsoidSrep(PyObject*, PyObject*)’:
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx:91:68: error: invalid static_cast from type ‘vtkObjectBase*’ to type ‘vtkBackwardFlowLogic*’
   vtkBackwardFlowLogic *op = static_cast<vtkBackwardFlowLogic *>(vp);
                                                                    ^
make[5]: *** [E/SRep/SkeletalRepresentationInitializer/Logic/CMakeFiles/vtkSlicerSkeletalRepresentationInitializerModuleLogicPythonD.dir/vtkBackwardFlowLogicPython.cxx.o] Error 1
make[4]: *** [E/SRep/SkeletalRepresentationInitializer/Logic/CMakeFiles/vtkSlicerSkeletalRepresentationInitializerModuleLogicPythonD.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [all] Error 2
make[2]: *** [slicersources-build/Slicer-prefix/src/Slicer-stamp/Slicer-build] Error 2
make[1]: *** [slicersources-build/CMakeFiles/Slicer.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -196,11 +196,12 @@ list(APPEND Slicer_EXTENSION_SOURCE_DIRS ${${extension_name}_SOURCE_DIR})
 #-----------------------------------------------------------------------------
 # Srep
 set(extension_name "SlicerSkeletalRepresentation")
-set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/SRep")
-FetchContent_Populate(${extension_name}
+set(short_extension_name "SRep") # Use shorter extension name to avoid issues with too long paths on windows
+set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/${short_extension_name}")
+FetchContent_Populate(${short_extension_name}
   SOURCE_DIR     ${${extension_name}_SOURCE_DIR}
-  GIT_REPOSITORY ${EP_GIT_PROTOCOL}://github.com/KitwareMedical/SlicerSkeletalRepresentation
-  GIT_TAG        027a1b8a85784ad55d08874318be46a0c76c6522 # initialize and visualize s-rep
+  GIT_REPOSITORY ${EP_GIT_PROTOCOL}://github.com/KitwareMedical/SlicerSkeletalRepresentation.git
+  GIT_TAG        7be23feb8bef52e9e319637c37acf68f12d73ceb
   GIT_PROGRESS   1
   QUIET
   )

